velox/exec/Window.cpp

```diff
@@ -629,12 +683,20 @@ RowVectorPtr Window::getOutput() {
     }
   }
 
+  if (!currentPartition_->isComplete() && currentPartition_->numRows() == 0) {
+    return nullptr;
+  }
+
   auto numOutputRows = std::min(numRowsPerOutput_, numRowsLeft);
   auto result = BaseVector::create<RowVector>(
       outputType_, numOutputRows, operatorCtx_->pool());
 
   // Compute the output values of window functions.
   auto numResultRows = callApplyLoop(numOutputRows, result);
+  if (currentPartition_) {
+    currentPartition_->clearOutputRows(numResultRows);
```

Comment:
Thinking this logic more we can do this only for RowsStreamingBuild... So lets add a variable in WindowPartition indicating if forPartialUse (set by RowsStreamingBuild in the WindowPartition constructor) and clear only if forPartialUse is set.

Reply:
@aditi-pandit Yes. Add partial_ variable to indicate the window partition is partial.